src/dotnet-aspnet-codegenerator/Program.cs

```diff
@@ -6,127 +6,218 @@
 
+        private static bool isProjectDependencyCommand;
+
         public static void Main(string[] args)
         {
+
             Stopwatch stopWatch = new Stopwatch();
             stopWatch.Start();
-
             _logger = new ConsoleLogger();
+            // We don't want to expose this flag, as it only needs to be used for identifying self invocation. 
+            isProjectDependencyCommand = args.Contains("--no-dispatch");
+            _logger.LogMessage($"Is Dependency COmmand: {isProjectDependencyCommand}");
+            try
+            {
+                Execute(args);
+            }
+            finally
+            {
+                stopWatch.Stop();
+                TimeSpan ts = stopWatch.Elapsed;
+                string elapsedTime = string.Format("{0:00}:{1:00}:{2:00}.{3:00}",
+                    ts.Hours, ts.Minutes, ts.Seconds,
+                    ts.Milliseconds / 10);
+                if (!isProjectDependencyCommand)
+                {
+                    // Check is needed so we don't show the runtime twice (once for the portable process and once for the dependency process)
+                    _logger.LogMessage("RunTime " + elapsedTime, LogMessageLevel.Information);
+                }
+            }
+        }
+
+        /// <summary>
+        /// The execution is done in 2 phases. 
+        /// Phase 1 ::
+        ///    1. Determine if the tool is running as a project dependency or not. 
+        ///    2. Try getting the project context for the project (use netcoreapp1.0 as the tfm if not running as dependency command or else use the tfm passed in)
+        ///    3. If not running as dependency command and project context cannot be built using netcoreapp1.0, invoke project dependency command with the first tfm found in the project.json
+        ///    
+        /// Phase 2 ::
+        ///     1. After successfully getting the Project context, invoke the CodeGenCommandExecutor.
+        /// </summary>
+        private static void Execute(string[] args)
+        {
             var app = new CommandLineApplication(false)
             {
                 Name = APPNAME,
                 Description = APP_DESC
             };
+
+            // Define app Options;
             app.HelpOption("-h|--help");
             var projectPath = app.Option("-p|--project", "Path to project.json", CommandOptionType.SingleValue);
             var packagesPath = app.Option("-n|--nugetPackageDir", "Path to check for Nuget packages", CommandOptionType.SingleValue);
             var appConfiguration = app.Option("-c|--configuration", "Configuration for the project (Possible values: Debug/ Release)", CommandOptionType.SingleValue);
-            
+            var framework = app.Option("-tfm|--targetFramework", "Target Framework to use. (Short folder name of the tfm. eg. net451)", CommandOptionType.SingleValue);
+            var buildBasePath = app.Option("-b|--buildBasePath", "", CommandOptionType.SingleValue);
```

Comment:
Shouldn't be pascal casing these params.

`nuget-package-dir`, `target-framework`, `build-base-path`